src/ripple/app/tests/common_ledger.cpp

```diff
@@ -0,0 +1,262 @@
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tests/common_ledger.h>
+
+namespace ripple {
+namespace test {
+
+Amount::Amount(double value_, std::string currency_, TestAccount issuer_)
+    : value(value_)
+    , currency(currency_)
+    , issuer(issuer_)
+{
+}
+
+Json::Value
+Amount::getJson() const
+{
+    Json::Value tx_json;
+    tx_json["currency"] = currency;
+    tx_json["issuer"] = issuer.pk.humanAccountID();
+    tx_json["value"] = std::to_string(value);
+    return tx_json;
+}
+
+// Helper function to parse a transaction in Json, sign it with account,
+// and return it as a STTx
+STTx
+parseTransaction(TestAccount& account, Json::Value const& tx_json, bool sign)
+{
+    STParsedJSONObject parsed("tx_json", tx_json);
+    std::unique_ptr<STObject> sopTrans = std::move(parsed.object);
+    if (sopTrans == nullptr)
+        throw std::runtime_error(
+        "sopTrans == nullptr");
+    sopTrans->setFieldVL(sfSigningPubKey, account.pk.getAccountPublic());
+    auto tx = STTx(*sopTrans);
+    if (sign)
+        tx.sign(account.sk);
+    return tx;
+}
+
+// Helper function to apply a transaction to a ledger
+void
+applyTransaction(Ledger::pointer const& ledger, STTx const& tx, bool check)
+{
+    TransactionEngine engine(ledger);
+    bool didApply = false;
+    auto r = engine.applyTransaction(tx, tapOPEN_LEDGER | (check ? tapNONE : tapNO_CHECK_SIGN),
+        didApply);
+    if (r != tesSUCCESS)
+        throw std::runtime_error(
+        "r != tesSUCCESS");
+    if (!didApply)
+        throw std::runtime_error(
+        "didApply");
+}
+
+// Create genesis ledger from a start amount in drops, and the public
+// master RippleAddress
+Ledger::pointer
+createGenesisLedger(std::uint64_t start_amount_drops, TestAccount const& master)
+{
+    Ledger::pointer ledger = std::make_shared<Ledger>(master.pk,
+        start_amount_drops);
+    ledger->updateHash();
+    ledger->setClosed();
+    if (!ledger->assertSane())
+        throw std::runtime_error(
+        "! ledger->assertSane()");
+    return ledger;
+}
+
+// Create an account represented by public RippleAddress and private
+// RippleAddress
+TestAccount
+createAccount(std::string const& passphrase)
+{
+    RippleAddress const seed
+        = RippleAddress::createSeedGeneric(passphrase);
+    RippleAddress const generator
+        = RippleAddress::createGeneratorPublic(seed);
+
+    return {
+        std::move(RippleAddress::createAccountPublic(generator, 0)),
+        std::move(RippleAddress::createAccountPrivate(generator, seed, 0)),
+        std::uint64_t(0)
+    };
+}
+
+void
+freezeAccount(TestAccount& account, Ledger::pointer const& ledger, bool sign)
+{
+    Json::Value tx_json;
+    tx_json["TransactionType"] = "AccountSet";
```

Comment:
Not your code, so you needn't pay attention to this.  But for these camel-case field names I've been using the `SField::getJsonName()` method.  So, for example, this might be `sfTransactionType.getJsonName()` instead of `"TransactionType"`.  Just a thought.